File: webapp.client/src/app/interfaces/create-product-request-model.ts
export interface CreateProductRequestModel {
  name: string;
  description: string;
  price: number;
  categorie: string;
  profileImage: File;
  createdOn : Date;
  isActive : boolean;
  isDeleted : boolean;
}

export interface UpdateProductRequestModel {
    id : string;
    name: string;
    description: string;
    price: number;
    categorie: string;
    profileImage: File;
    createdOn : Date;
    isActive : boolean;
    isDeleted : boolean;
  }

File: WebApp.Model/Product/CreateProductRequestModel.cs
using Microsoft.AspNetCore.Http;

namespace WebApp.Model.Product
{
    public class CreateProductRequestModel
    {
        public string Name { get; set; } = string.Empty;
        public float Price { get; set; }
        public string Description { get; set; } = string.Empty;
        public IFormFile? ProfileImage { get; set; }
        public string Categorie { get; set; } = string.Empty;
        public DateTime CreatedOn { get; set; }
        public bool IsActive { get; set; }
        public bool IsDeleted { get; set; }
    }
}
